scripts/partition_manager.py

```diff
@@ -372,10 +372,26 @@ def _set_addresses_and_align(reqs, sub_partitions, solution, size, start, dynami
         if align_if_required(i, dynamic_partitions, False, reqs, solution):
             _set_addresses_and_align(reqs, sub_partitions, solution, size, start, dynamic_partitions)
 
+def align_is_valid(align):
+    if 'start' in align:
+        if isinstance(align['start'],list) and not (isinstance(x,int) for x in align['start']):
+            print(f"elements in {align['start']} is not type of {int}")
+            raise TypeError
+        elif not isinstance(align['start'], int):
+            print(f"{align['start']} is not type of {int}")
+            raise TypeError
+    if 'end' in align:
+        if isinstance(align['end'],list) and not (isinstance(x,int) for x in align['end']):
+            print(f"elements in {align['end']} is not type of {int}")
+            raise TypeError
```

Comment:
Why do the printing like this, instead of:
```suggestion
raise TypeError(f"elements in {align['end']} is not type of {int}")
```
as that will ensure the print to be at bottom of stack trace, as:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/projects/github/ncs/nrf/scripts/partition_manager.py", line 389, in align_is_valid
    raise TypeError(f"{align['end']} is not type of {int}")
TypeError: ['CONFIG_VAR'] is not type of <class 'int'>
```

instead of current solution:
```
['CONFIG_VAR'] is not type of <class 'int'>
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/projects/github/ncs/nrf/scripts/partition_manager.py", line 389, in align_is_valid
    raise TypeError
```

Reply:
I like it at the top of the stack trace :D But sure I can put it at the bottom :+1: